### PR TITLE
Fixes support for video sequences bigger than 2GB

### PR DESCRIPTION
**Pull request by JanuszL**
- fixes wrong pointer arithmetic based on int that leads to overflow
  and illegal memory access  when the video sequence length is bigger than 2GB
- relates https://github.com/NVIDIA/DALI/issues/3542

## Category:
**Bug fix** (*non-breaking change which fixes an issue*)

## Description:
- fixes wrong pointer arithmetic based on int that leads to overflow
  and illegal memory access  when the video sequence length is bigger than 2GB

### Affected modules and functionalities:
- video reader and video reader resize

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
    - test_video_pipeline.test_2gb_sequence
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: dali/test/python/test_video_pipeline.py [hunk 1]
```diff
@@ -628,3 +628,24 @@ def video_decoder_pipeline():
         b = b.as_cpu().as_array()
         absdiff = np.abs(a.astype(int) - b.astype(int))
         assert np.mean(absdiff) < 2, f"error is {np.mean(absdiff)}"
+
+
+def test_2gb_sequence():
+    # make sure the sequence extends 2GB
```
Comment (mzient) [inline dali/test/python/test_video_pipeline.py:634]: Nitpick
```suggestion
    # make sure the sequence exceeds beyond 2GB
```
or
```suggestion
    # make sure the sequence extends beyond 2GB
```

**Review by mzient — APPROVED** [review]

File: dali/test/python/test_video_pipeline.py (hunk 1 above)
Comment (JanuszL) [reply dali/test/python/test_video_pipeline.py]: Fixed

**Comment by JanuszL** [comment]
!build

File: dali/test/python/test_video_pipeline.py [hunk 2]
```diff
@@ -628,3 +628,24 @@ def video_decoder_pipeline():
         b = b.as_cpu().as_array()
         absdiff = np.abs(a.astype(int) - b.astype(int))
         assert np.mean(absdiff) < 2, f"error is {np.mean(absdiff)}"
+
+
+def test_2gb_sequence():
```
Comment (jantonguirao) [inline dali/test/python/test_video_pipeline.py:633]: how costly is to run this? Should we mark it as "slow"?

Comment (JanuszL) [reply dali/test/python/test_video_pipeline.py:633]: CI shows `[success] 4.29% test_video_pipeline.test_2gb_sequence: 2.0841s`
Do you think it is much?

Comment (jantonguirao) [reply dali/test/python/test_video_pipeline.py:633]: I think we can stay. But those slower tests often become a problem in sanitizer builds where everything runs much slower. Let's leave it an see how it goes

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**